From: Stained Glass OS <ke7oxh@gmail.com>
Subject: [PATCH] taskschd: DllGetVersion (3515)

taskschd: DllGetVersion

Fills DLLVERSIONINFO and DLLVERSIONINFO2 (10.0.19041, NT platform) and rejects other structure sizes.
---
--- a/dlls/taskschd/taskschd.c
+++ b/dlls/taskschd/taskschd.c
@@ -27,6 +27,7 @@
 #include "objbase.h"
 #include "oleauto.h"
 #include "rpcproxy.h"
+#include "shlwapi.h"
 #include "taskschd.h"
 #include "taskschd_private.h"
 
@@ -347,3 +348,27 @@
     FIXME("class %s/%s is not implemented\n", debugstr_guid(rclsid), debugstr_guid(riid));
     return CLASS_E_CLASSNOTAVAILABLE;
 }
+
+/***********************************************************************
+ *      DllGetVersion
+ */
+HRESULT WINAPI DllGetVersion(DLLVERSIONINFO *info)
+{
+    TRACE("%p\n", info);
+
+    if (!info || (info->cbSize != sizeof(DLLVERSIONINFO) && info->cbSize != sizeof(DLLVERSIONINFO2)))
+        return E_INVALIDARG;
+
+    info->dwMajorVersion = 10;
+    info->dwMinorVersion = 0;
+    info->dwBuildNumber = 19041;
+    info->dwPlatformID = DLLVER_PLATFORM_NT;
+    if (info->cbSize == sizeof(DLLVERSIONINFO2))
+    {
+        DLLVERSIONINFO2 *info2 = (DLLVERSIONINFO2 *)info;
+
+        info2->dwFlags = 0;
+        info2->ullVersion = MAKEDLLVERULL(10, 0, 19041, 0);
+    }
+    return S_OK;
+}
--- a/dlls/taskschd/taskschd.spec
+++ b/dlls/taskschd/taskschd.spec
@@ -1,5 +1,5 @@
 @ stdcall -private DllCanUnloadNow()
 @ stdcall -private DllGetClassObject(ptr ptr ptr)
-@ stub DllGetVersion
+@ stdcall -private DllGetVersion(ptr)
 @ stdcall -private DllRegisterServer()
 @ stdcall -private DllUnregisterServer()
